platforms/android/tangram/src/main/java/com/mapzen/tangram/MapController.java

```diff
@@ -1089,24 +1098,56 @@ public void removeAllMarkers() {
      */
     public void setMapChangeListener(@Nullable final MapChangeListener listener) {
         mapChangeListener = listener;
+        mapRegionChangeStateManager = new MapRegionChangeStateManager(mapChangeListener);
     }
 
-    //Convenience member functions
-    private void onRegionWillChange(boolean animated) {
-        if (mapChangeListener != null) {
-            mapChangeListener.onRegionWillChange(animated);
+    private static class MapRegionChangeStateManager {
```

Comment:
Why create an inner class here? Seems to create clutter and I'm not sure what's gained from it.

Reply:
I wanted to encapsulate the logic and not make it part of MapController itself. Since its internal, maybe I can add a package private external class for the same.

Reply:
There's not much to encapsulate. I think the worst that we could do is accidentally set the state directly instead of using the method - and actually we can do that anyway even with the inner class.